### DISCUSSION TITLE
TOUGH2/TOUGH3

### DISCUSSION THREAD

**Discussion by SHerrboldt** [NatLabRockies/GEOPHIRES-X#289] [Q&A] [answered]
I have been trying to get GEOPHIREs to use TOUGH2/TOUGH3 and it is not working. The error I keep getting is:

'executable file does not exist in current working directory'

I have tried placing the executable file, along with the input file, in the same folder as my Python code. This has not fixed the error. I am aware that both the input and TOUGH executable files need to be called in the input but I have not been able to get past this error.
If anybody has any insight as to how I could fix this issue or share how you are able to successfully use TOUGH in GEOPHIREs.

**Comment by softwareengineerprogrammer** [comment]
@SHerrboldt Is your executable named `xt2_eo1.exe` ([the name expected by GEOPHIRES in source code](https://github.com/NREL/GEOPHIRES-X/blob/3fb5b63676735f2696ddb5ce419eca8091999f54/src/geophires_x/TOUGH2Reservoir.py#L129-L135))?

**Reply by SHerrboldt** [reply]
I changed that to match with my executable (tough3-eos1.exe).

**Reply by softwareengineerprogrammer** [reply]
Change relevant lines to the following and verify the full path is as you expect:

```python
        path_to_exe = str('xt2_eos1.exe')
        exe_full_path = os.path.join(os.getcwd(), path_to_exe)
        print(f'TOUGH executable file full path: {exe_full_path}')
        if not os.path.exists(exe_full_path):
```

**Reply by SHerrboldt** [reply]
I made the relevant changes and was able to find a path:

     TOUGH executable file full path: C:\Users\101062015\my-geophires-project\tough3-eos1.exe

This is not the directory from which I am running the start.py file. I tried running the file with the TOUGH executable in the directory above and am getting the same error. 

Is there a way to specify in the code the exact path I want to use for the TOUGH executable?

**Reply by softwareengineerprogrammer** [reply]
Doesn't look like there's a way to do so currently, so I'd just go ahead and hardcode the path locally in code for now. I've filed a tracking issue to add support for a custom executable path so eventually you won't have to do that: https://github.com/NREL/GEOPHIRES-X/issues/291?title=Parameter+for+TOUGH+executable+location

**Reply by SHerrboldt** [reply, marked as answer]
After the changes, I can run TOUGH executables. You have to rename the executable from inside of 'TOUGH2Reservoir.py'. Ctrl+f and search 'eos' to find it (found on line 37): 
  
   'xt2_eos1.exe'

By doing this, I can read my TOUGH executable and it works.
